AsyncDisplayKit/Layout/ASLayout.h

```diff
@@ -17,124 +17,117 @@
+ */
+extern ASLayout *ASCalculateRootLayout(id<ASLayoutable> rootLayoutable, const ASSizeRange sizeRange);
+
+/**
+ * Safely computes the layout of the given node by guarding against nil nodes.
+ * @param component The component to calculate the layout for.
+ * @param sizeRange The size range to calculate the node layout within.
+ * @param parentSize The parent size of the node to calculate the layout for.
+ */
+extern ASLayout *ASCalculateLayout(id<ASLayoutable> layoutable, const ASSizeRange sizeRange, const CGSize parentSize);
+
+ASDISPLAYNODE_EXTERN_C_END
+
+/**
  * A node in the layout tree that represents the size and position of the object that created it (ASLayoutable).
  */
 @interface ASLayout : NSObject
 
 /**
  * The underlying object described by this layout
  */
-@property (nonatomic, weak, readonly) id<ASLayoutable> layoutableObject;
+@property (nonatomic, weak, readonly) id<ASLayoutable> layoutable;
 
 /**
  * The type of ASLayoutable that created this layout
  */
-@property (nonatomic, readonly) ASLayoutableType type;
+@property (nonatomic, assign, readonly) ASLayoutableType type;
 
 /**
  * Size of the current layout
  */
-@property (nonatomic, readonly) CGSize size;
+@property (nonatomic, assign, readonly) CGSize size;
 
 /**
  * Position in parent. Default to CGPointNull.
  * 
  * @discussion When being used as a sublayout, this property must not equal CGPointNull.
  */
-@property (nonatomic, readwrite) CGPoint position;
-
-/**
- * The size range that was use to determine the size of the layout.
- */
-@property (nonatomic, readonly) ASSizeRange constrainedSizeRange;
+@property (nonatomic, assign, readwrite) CGPoint position;
 
 /**
  * Array of ASLayouts. Each must have a valid non-null position.
  */
-@property (nonatomic, readonly) NSArray<ASLayout *> *sublayouts;
-
-/**
- * Mark the layout dirty for future regeneration.
- */
-@property (nonatomic, getter=isDirty) BOOL dirty;
+@property (nonatomic, copy, readonly) NSArray<ASLayout *> *sublayouts;
 
 /**
  * @abstract Returns a valid frame for the current layout computed with the size and position.
  * @discussion Clamps the layout's origin or position to 0 if any of the calculated values are infinite.
  */
-@property (nonatomic, readonly) CGRect frame;
+@property (nonatomic, assign, readonly) CGRect frame;
 
 /**
  * Designated initializer
  */
-- (instancetype)initWithLayoutableObject:(id<ASLayoutable>)layoutableObject
-                    constrainedSizeRange:(ASSizeRange)sizeRange
-                                    size:(CGSize)size
-                                position:(CGPoint)position
-                              sublayouts:(NSArray *)sublayouts NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithLayoutable:(id<ASLayoutable>)layoutable
+                              size:(CGSize)size
+                          position:(CGPoint)position
+                        sublayouts:(NSArray *)sublayouts NS_DESIGNATED_INITIALIZER;
```

Comment:
Question: Can we improve the type declaration of `sublayouts` to `NSArray<ASLayout *> *`?